test-utils/src/rpc/solana_rpc.rs

```diff
@@ -269,15 +265,18 @@ impl RpcConnection for SolanaRpcConnection {
             .client
             .request_airdrop(to, lamports)
             .map_err(RpcError::from)?;
-        let confirmed = self
-            .client
-            .confirm_transaction_with_commitment(&signature, CommitmentConfig::confirmed())?
-            .value;
-        if confirmed {
-            Ok(signature)
-        } else {
-            Err(RpcError::CustomError("Airdrop failed".to_string()))
+
+        loop {
+            let confirmed = self
+                .client
+                .confirm_transaction_with_commitment(&signature, self.client.commitment())?
+                .value;
+            if confirmed {
+                break;
+            }
         }
```

Comment:
Also no need to do that in this pr but we need to get rid of these loops without a timeout.
If the transaction fails this will result in an infinite loop.

Reply:
Hm, as far as I understand, If the transaction fails, it will break the loop due to the error propagation through `?`. 
The only theoretical possibility of an infinite loop: transaction never confirmed.

Also, https://github.com/anza-xyz/agave/blob/master/rpc-client/src/rpc_client.rs#L979-L985

Reply:
I have been stuck in a loop like this when testing a couple days ago.
In the case where I was stuck, the rpc returned transaction signature even though tx has failed and I was able to look at the failed tx in the explorer.

Reply:
Try this logic with transaction that does a transfer from an account with insufficient funds.

Reply:
Gotcha, thanks. I wrote a note for myself, let's fix this on another PR.